apps/web/vite.config.ts

```diff
@@ -27,11 +30,39 @@ const knownIcons = [
 
 export default defineConfig(({ isSsrBuild }) => ({
   build: {
-    rollupOptions: isSsrBuild
-      ? {
-          input: './workers/app.ts',
-        }
-      : undefined,
+    rollupOptions: {
+      ...(isSsrBuild
+        ? {
+            input: './workers/app.ts',
+          }
+        : undefined),
+      plugins: [
+        /**
+         * Enables Rollup (prod) Buffer support (vite-plugin-node-polyfills doesn't work)
+         * @see https://github.com/vitejs/vite/discussions/6180#discussioncomment-7254293
+         */
+        inject({ Buffer: ['buffer/', 'Buffer'] }),
+        nodePolyfills({
+          include: 'buffer',
+        }),
+      ],
+    },
+  },
+  optimizeDeps: {
+    esbuildOptions: {
+      define: {
+        global: 'globalThis',
+      },
+      /**
+       * Enables Esbuild (dev) Buffer support (vite-plugin-node-polyfills doesn't work)
+       * @see https://github.com/vitejs/vite/discussions/6180#discussioncomment-7254293
+       */
+      plugins: [
+        NodeGlobalsPolyfillPlugin({
+          buffer: true,
+        }),
```

Comment:
Out of curiosity, did you try a classic polyfil by just doing 


```ts
import { Buffer } from 'buffer'
globalThis.Buffer = Buffer
```

that'd be much nicer than having to change all the build process for this

Reply:
I get an err trying this
`app/root.tsx (1:9): "Buffer" is not exported by "__vite-browser-external", imported by "app/root.tsx".`
seems like under the hood vite changes original buffer to it’s own empty one, quite weird

Reply:
and using a different lib like https://www.npmjs.com/package/safe-buffer ?

Reply:
yeah, works now